devdoc/thandle_requirements.md

```diff
@@ -223,4 +223,20 @@ void THANDLE_MOVE(T)(THANDLE(T) * t1, THANDLE(T) * t2)
 
 **SRS_THANDLE_02_037: [** If `*t1` is not `NULL` and `*t2` is `NULL` then `THANDLE_MOVE` shall `THANDLE_DEC_REF` `*t1`, set `*t1` to `NULL` and return. **]**
 
-**SRS_THANDLE_02_038: [** If `*t1` is not `NULL` and `*t2` is not `NULL` then `THANDLE_MOVE` shall `THANDLE_DEC_REF` `*t1`, set `*t1` to `*t2`, set `*t2` to `NULL` and return. **]**
\ No newline at end of file
+**SRS_THANDLE_02_038: [** If `*t1` is not `NULL` and `*t2` is not `NULL` then `THANDLE_MOVE` shall `THANDLE_DEC_REF` `*t1`, set `*t1` to `*t2`, set `*t2` to `NULL` and return. **]**
+
+###  THANDLE_INITIALIZE_MOVE_MACRO(T)
+```c
+THANDLE_INITIALIZE_MOVE_MACRO(T)
+void THANDLE_INITIALIZE_MOVE(T)(THANDLE(T) * t1, THANDLE(T) * t2)
+```
+
+`THANDLE_INITIALIZE_MOVE` moves `*t2` under `*t1` and NULLs *t2, assuming that `t1` is not initialized. 
+
+**SRS_THANDLE_01_001: [** If `t1` is `NULL` then `THANDLE_INITIALIZE_MOVE` shall return. **]**
+
+**SRS_THANDLE_01_002: [** If `t2` is `NULL` then `THANDLE_INITIALIZE_MOVE` shall return. **]**
+
+**SRS_THANDLE_01_003: [** If `*t2` is `NULL` then `THANDLE_INITIALIZE_MOVE` shall `THANDLE_DEC_REF` `*t1`, set `*t1` to `NULL` and return. **]**
```

Comment:
>`THANDLE_DEC_REF` `*t1`, [](start = 82, length = 24)

shouldn't dec_ref

Reply:
Spec fixed, we should not dec ref anything.

---
In reply to: [470882740](https://github.com/Azure/azure-c-util/pull/60#discussion_r470882740) [](ancestors = 470882740)